### PR TITLE
Uploads can be done to any storage provider

### PR DESCRIPTION
**Pull request by mnaamani**
Implements https://github.com/Joystream/joystream/issues/2269

Building on https://github.com/Joystream/joystream/pull/2266 and https://github.com/Joystream/joystream/pull/2272

Any storage provider can accept content upload. Upload can be repeated at any time, which also means that if content is ever lost by all nodes, anyone can re-upload the content.

Dropped 'Rejected' enum variant of LiaisonJudgement

### CODE REVIEW

File: storage-node/packages/cli/src/commands/download.ts [hunk 1]
```diff
@@ -5,38 +5,27 @@ import { BaseCommand } from './base'
...
   }
 
   // Shows command usage. Overrides the abstract method from the base class.
   protected showUsage() {
     console.log(
       chalk.yellow(`
-        Usage:   storage-cli download colossusURL contentID filePath
-        Example: storage-cli download http://localhost:3001 0x7a6ba7e9157e5fba190dc146fe1baa8180e29728a5c76779ed99655500cff795 ./movie.mp4
+        Usage:   storage-cli download contentID filePath
+        Example: storage-cli download 0x7a6ba7e9157e5fba190dc146fe1baa8180e29728a5c76779ed99655500cff795 ./movie.mp4
```
Comment (shamil-gadelshin) [inline storage-node/packages/cli/src/commands/download.ts]: Please, change an example for download to different CID format (eg.: "5GFwJXfTf19eUAgoHLsDwxd3YZ5ADdHs5mSzPZVU3Aw6T5Gf")

File: runtime-modules/storage/src/data_directory.rs [hunk 2]
```diff
@@ -818,7 +798,8 @@ impl<T: Trait> Module<T> {
                 size: content.size,
                 added_at: common::current_block_time::<T>(),
                 owner: owner.clone(),
-                liaison,
+                // It is irrelevant what liaison is set when judgement is pending
```
Comment (shamil-gadelshin) [inline runtime-modules/storage/src/data_directory.rs]: It is confusing. Do you mind changing it to the Option?

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Looks good. I tested "upload" and "download" commands. Left just a couple of small issues.

File: storage-node/packages/cli/src/commands/download.ts (hunk 1 above)
Comment (mnaamani) [reply storage-node/packages/cli/src/commands/download.ts]: makes sense, will update

File: runtime-modules/storage/src/data_directory.rs (hunk 2 above)
Comment (mnaamani) [reply runtime-modules/storage/src/data_directory.rs]: I agree, an Option will be clearer.

File: storage-node/packages/cli/src/commands/download.ts (hunk 1 above)
Comment (mnaamani) [reply storage-node/packages/cli/src/commands/download.ts]: done in https://github.com/Joystream/joystream/pull/2278/commits/edf7e2671e3e897505158fdedefb7892c24fcbb1

File: runtime-modules/storage/src/data_directory.rs (hunk 2 above)
Comment (mnaamani) [reply runtime-modules/storage/src/data_directory.rs]: done in https://github.com/Joystream/joystream/pull/2278/commits/fad04b7e2caf78333f360ba04741c0b92d9f5725

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Pull request merged**